app/src/parser_impl.c

```diff
@@ -282,6 +290,9 @@ parser_error_t parse_additional_typebytes(parser_context_t *ctx, uint8_t type, u
         }
 
         default : {
+            // we support now generic arguments
+            // in transactions but we only support
+            // the types define above
             return parser_unexpected_type;
         }
```

Comment:
I would consider making a different error-code for this? Since we use parser_unexpected_type for actual errors too (see _read in parser.c). I think it's best to have it a unique code.

Reply:
Hopefully this unique code enables to get rid of the below check?

else if (err != parser_runtimearg_notfound && err != parser_unexpected_type)

Reply:
oh, yes, makes sense..